src/Dependencies/DependencyList.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace DepDoc\Dependencies;
+
+use phpDocumentor\Reflection\Types\Boolean;
+
+class DependencyList implements \ArrayAccess
+{
+    /** @var DependencyData[] */
+    protected $dependencies = [];
+
+    /**
+     * @param DependencyData $data
+     * @return DependencyList
+     */
+    public function add(DependencyData $data): DependencyList
+    {
+        $this[$this->getListKey($data->getPackageManagerName(), $data->getPackageName())] = $data;
+
+        return $this;
+    }
+
+    /**
+     * @param string $packageManagerName
+     * @param string $packageName
+     * @return bool
+     */
+    public function has(string $packageManagerName, string $packageName): bool
+    {
+        return $this->offsetExists($this->getListKey($packageManagerName, $packageName));
+    }
+
+    /**
+     * @param string $packageManagerName
+     * @param string $packageName
+     * @return null|DependencyData
+     */
+    public function get(string $packageManagerName, string $packageName): ?DependencyData
+    {
+        if (!$this->has($packageManagerName, $packageName)) {
+            return null;
+        }
+
+        return $this[$this->getListKey($packageManagerName, $packageName)];
+    }
+
+    /**
+     * @param string $packageManagerName
+     * @param string $packageName
+     * @return string
+     */
+    public function getListKey(string $packageManagerName, string $packageName): string
+    {
+        return sprintf('%s-%s', $packageManagerName, $packageName);
+    }
+
+    /**
+     * Whether a offset exists
+     * @link http://php.net/manual/en/arrayaccess.offsetexists.php
+     * @param mixed $offset <p>
+     * An offset to check for.
+     * </p>
+     * @return boolean true on success or false on failure.
+     * </p>
+     * <p>
+     * The return value will be casted to boolean if non-boolean was returned.
+     * @since 5.0.0
+     */
+    public function offsetExists($offset): bool
```

Comment:
There seems to be some copy pasta in here. Are this functions needed because you implemented ``\ArrayAccess``?